npc/re/jobs/2e/rebellion.txt

```diff
@@ -0,0 +1,1093 @@
+}
+
+job_gun,130,146,0	script	shootingrange#5	WARPNPC,3,3,{
+	end;
+OnTouch:
+	warp("job_gun", 120, 130);
+	end;
+}
+
+job_gun,165,18,1	script	Furnace Controller#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != G_L_SHECIL) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	if (questprogress(12342) != 1) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	mes("Do you want to dredge the mold up from the furnace?");
+	next;
+	if (select("Quit.", "Begin.") == 1) {
+		mes("Switch to the Standby mode.");
+		close;
+	}
+	hideonnpc("Furnace Controller#1");
+	donpcevent("Furnace Controller#2::OnStart");
+	mes("Initializing Furnace Controller ZX-3100...");
+	close;
+
+OnInit:
+	hideoffnpc("Furnace Controller#1");
+	end;
+}
+
+job_gun,214,36,1	script	Auto Anvil#1	CLEAR_NPC,{
+	if (Class != Job_Gunslinger) {
+		mes("Only authorized user can do the operation.");
+		close;
+	}
+	if (countitem(Tiny_Flame) > 0) {
+		mes("Unauthorized firearm creation has been detected. Report to Manager Elwin.");
+		close;
+	}
+	if (getstatus(SC_MONSTER_TRANSFORM, 1) != 2431 || countitem(Steel_Article_) <= 4 || questprogress(12342) != 1) {
+		mes("Not an authorized user. Abort the operation.");
+		close;
+	}
+	
+	mes("Do you want to use the auto anvil and shape the standard Steel Artifacts into a firearm?");
+	next;
+	if (select("Quit.", "Begin.") == 1) {
+		mes("For your safety, please step away from the auto anvil.");
+		close;
+	}
+	.@ma_wp = rand(1, 10);
+	if (.@ma_wp > 6) {
+		specialeffect(EF_REPAIRWEAPON);
+		mes("You have successfully created a firearm using the auto anvil. ^ff0000Please report before your transformation expires^000000.");
+		delitem(Steel_Article_, 5);
+		getitem(Tiny_Flame, 1);
+	} else {
+		specialeffect(EF_SUI_EXPLOSION);
+		mes("Failed to create a firearm.");
+		delitem(Steel_Article_, 5);
+	}
+	close;
+	
+OnInit:
+	hideoffnpc("Auto Anvil#1");
+	end;
+}
+
+job_gun,165,18,1	script	Furnace Controller#2	CLEAR_NPC,{
+	mes("The system is busy dredging a completed mold from the furnace.");
+	close;
+
+OnInit:
+	hideonnpc("Furnace Controller#2");
+	end;
+
+OnStart:
+	hideoffnpc("Furnace Controller#2");
+	sleep(3000);
+	mapannounce("job_gun", "Furnace Controller: Dredging a mold up from the furnace... Workers, stand by at the production line.", bc_map, "0x00ff44");
+	sleep(500);
+	.@wh_start = 0;
+	while (true) {
+		if (.@wh_start >= 30)
+			break;
+		
+		.@eff_rnd = rand(1, 100);
+		if (.@eff_rnd < 33) { // 0 - 32
+			donpcevent("Heating Furnace#3::OnStart");
+		} else if (.@eff_rnd > 66) { // 67-100
+			donpcevent("Heating Furnace#4::OnStart");
+		} else { // 33-66
+			donpcevent("Heating Furnace#5::OnStart");
+		}
+		sleep(200);
+		.@d_itemx = rand(127, 143);
+		.@d_itemy = rand(26, 29);
+		makeitem(Steel_Article, 1, "job_gun", .@d_itemx, .@d_itemy);
+		.@scc_rnd = rand(1, 10);
+		if (.@scc_rnd > 7) {
```

Comment:
```suggestion
		if (rand(1, 10) > 7) {
```
and remove `.@scc_rnd = rand(1, 10);`

Reply:
done, will be pushed together in next commit